srecord/input/file/needham.h

```diff
@@ -67,7 +67,7 @@ class input_file_needham:
     input_file_needham(const std::string &file_name);
 
     auto read_inner(record &) -> bool;
-    bool garbage_warning;
+    bool garbage_warning{};
```

Comment:
In principal, I like the check. Curious about why it triggered on garbage_warning but not on the following seen_some_input.

Reply:
Probably: `seen_some_input` seems to be unused and should be removed.